libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -51,10 +55,12 @@ class RustMatrixAuthenticationService @Inject constructor(
     private val sessionStore: SessionStore,
     userAgentProvider: UserAgentProvider,
     private val rustMatrixClientFactory: RustMatrixClientFactory,
+    private val databaseKeysProvider: DatabaseKeysProvider,
+    private val buildMeta: BuildMeta,
 ) : MatrixAuthenticationService {
     private val authService: RustAuthenticationService = RustAuthenticationService(
         basePath = baseDirectory.absolutePath,
-        passphrase = null,
+        passphrase = getDatabasePassphrase(),
```

Comment:
We can't really place the passphrase in the `SessionData` as [per iOS](https://github.com/element-hq/element-x-ios/pull/2328) since we need it here. Not sure how they managed this on iOS

Reply:
OK, all of this above is fixed now, I have rewrite the change.